packages/block-library/src/cover/edit.js

```diff
@@ -264,6 +266,47 @@ function CoverEdit( {
 	const onSelectMedia = attributesFromMedia( setAttributes );
 	const isBlogUrl = isBlobURL( url );
 
+	const [ prevMinHeightValue, setPrevMinHeightValue ] = useState( minHeight );
```

Comment:
I guess these two state values could share the same useState call the value being an object with properties minHeight and minHeightUnit that could be directly passed to set attributes. These change also aligns with probable change we will have to do make minHeightUnit and minHeight a single attribute (a string that can contain multiple CSS valid properties).

If we don't initiate the state of previous to minHeight, minHeightUnit e.g: keep them undefined and also follow the suggestion of setting the previous values inside toggleMinFullHeight we can also probably remove the need for this condition bellow "// If there aren't previous values, take the default ones.".

Reply:
Nice suggestion. Going to refactoring using an object.

> These change also aligns with probable change we will have to do make minHeightUnit and minHeight a single attribute (a string that can contain multiple CSS valid properties).

💯 agree.
I did this approach [in the Background Size implementation](https://github.com/WordPress/gutenberg/pull/26976). The data is stored in the [backgroundSize attribute (string)](https://github.com/WordPress/gutenberg/blob/bf596cc512f411b9b2d8991db2f3499a537c7e66/packages/block-library/src/cover/block.json#L20-L23)ready to be used in the [inline style](https://github.com/WordPress/gutenberg/blob/bf596cc512f411b9b2d8991db2f3499a537c7e66/packages/block-library/src/cover/edit.js#L486). The block takes over of parsing and converting from/to this attribute to get the data usable by the UI components.

Happy to do this in a follow-up PR.